### PR TITLE
Feature/implement persistence layer

### PR DESCRIPTION
**Pull request by ElRaptorus**
## What did you change?

Implements a repository that uses sequelize for storing and querying `FlowNodeInstance`s and corresponding `ProcessToken`s.

## How can others test the changes?

Use the repository to make queries for flow node instances.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

**Comment by ElRaptorus** [comment]
https://github.com/process-engine-pro/organization/issues/152

File: src/schemas/process_token.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import * as Sequelize from 'sequelize';
+
+export interface IProcessTokenAttributes {
+  id: string;
+  processInstanceId: string;
+  processModelId: string;
+  correlationId: string;
+  identity: string; // Contains the stringified IIdentity
```
Comment (cmg-dev) [inline src/schemas/process_token.ts]: >  // Contains the stringified IIdentity

ist iwie nutzlos der kommentar.

File: src/schemas/process_token.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  createdAt: Date;
+  caller?: string; // Only set, if the process token belongs to a subprocess
+  payload: string; // Contains the stringified payload
```
Comment (cmg-dev) [inline src/schemas/process_token.ts]: same.

File: src/schemas/process_token.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import * as Sequelize from 'sequelize';
+
+export interface IProcessTokenAttributes {
+  id: string;
+  processInstanceId: string;
+  processModelId: string;
+  correlationId: string;
+  identity: string; // Contains the stringified IIdentity
+  createdAt: Date;
+  caller?: string; // Only set, if the process token belongs to a subprocess
```
Comment (cmg-dev) [inline src/schemas/process_token.ts]: sowas ist sinnvoll :)

**Comment by ElRaptorus** [comment]
@cmg-dev done

File: src/flow_node_instance_repository.ts [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+  }
+
+  public async persistOnExit(newProcessToken: Runtime.Types.ProcessToken,
+                             flowNodeId: string,
+                             flowNodeInstanceId: string): Promise<Runtime.Types.FlowNodeInstance> {
+
+    const matchingFlowNodeInstance: FlowNodeInstanceModel = await this.flowNodeInstanceModel.findOne({
+      where: {
+        flowNodeId: flowNodeId,
+        instanceId: flowNodeInstanceId,
```
Comment (5ebastianMeier) [inline src/flow_node_instance_repository.ts]: Please don't shorten the names of it's not absolutely necessary.

I would rather use `flowNodeInstanceId` than `instanceId`.

File: src/schemas/flow_node_instance.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import * as Sequelize from 'sequelize';
+import {ProcessToken} from './process_token';
+
+export interface IFlowNodeInstanceAttributes {
+  id: string;
+  instanceId: string;
```
Comment (5ebastianMeier) [inline src/schemas/flow_node_instance.ts]: see above

File: src/schemas/flow_node_instance.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  flowNodeId: string;
+  isSuspended: boolean;
+  // Contains the association to the ProcessToken model.
+  // Must be optional, otherwise this property will be expected in the attribute payload of `sequelize.define`.
+  processToken?: ProcessToken;
+}
+
+export type FlowNodeInstance = Sequelize.Instance<IFlowNodeInstanceAttributes> & IFlowNodeInstanceAttributes;
```
Comment (5ebastianMeier) [inline src/schemas/flow_node_instance.ts]: good type usage!

File: src/types.d.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import {DataTypeAbstract, DefineAttributeColumnOptions} from 'sequelize';
+
+type SequelizeAttribute = string | DataTypeAbstract | DefineAttributeColumnOptions;
+
+declare global {
```
Comment (5ebastianMeier) [inline src/types.d.ts]: A comment with a short example of what this snippet enables would be good for documentation purposes.

**Comment by ElRaptorus** [comment]
@5ebastianMeier done

**Pull request merged**